README.md

```diff
@@ -10,6 +10,75 @@ Release: [![Master](https://api.travis-ci.org/untangled-web/untangled-server.svg
+Next you will need to set your `:config-path` in `untangled-make-server` to be a reasonable value for development and potentially production. Note: this can be parameterized, so you can do things like have different paths in development and production.
+
+Example [system.clj](https://github.com/untangled-web/untangled-todomvc/blob/master/src/server/todomvc/system.clj#L15) & [defaults.edn](https://github.com/untangled-web/untangled-todomvc/blob/master/resources/config/defaults.edn) from untangled-todomvc.
+
+### Component Injection
+
+To inject [components](https://github.com/stuartsierra/component) into the untangled-server's system, add a `:components` map to `make-untangled-server` with keys being the name of the component in your `:parser-injections`, and the value being the component you created with `defrecord` or some function to inject depencies with `component/using`.
+Eg:
+```
+(declare build-hooks)
+
+(defrecord Database [conn]
+  component/Lifecycle
+  (start [this] ...)
+  (stop [this] ...))
+  
+(core/make-untangled-server
+  ...
+  :components {:db (map->Database {})
+               :hooks (build-hooks)})
+```
+see [Ring Handler Injection](https://github.com/untangled-web/untangled-server/tree/feature/documentation#ring-handler-injection) for the `build-hooks` implementation
+
+### Ring Handler Injection
+
+NOTE: May be subject to change/improvement
+
+There are two locations in untangled-server's pre-built handler stack, [pre-hook](https://github.com/untangled-web/untangled-server/blob/8dba26aafe36a5f0dab36d0dc89a98f43212df1d/src/untangled/server/impl/components/handler.clj#L176) and [fallback-hook](https://github.com/untangled-web/untangled-server/blob/8dba26aafe36a5f0dab36d0dc89a98f43212df1d/src/untangled/server/impl/components/handler.clj#L170), that are made publically accessible.
+The first step is to create a component that depends (`component/using`) on the `:handler`, and then on start to get and set the desired hook.
+```
+(defrecord Hooks [handler]
+  component/Lifecycle
+  (start [this]
+    (let [pre-hook (h/get-pre-hook handler)]
+      (h/set-pre-hook! handler
+        (comp
+          ... your-wrap-handlers-here ...
+          pre-hook 
+          ...or-here...)))))
+(defn build-hooks []
+  (component/using
+    (map->Hooks {})
+    [:handler]))
+```
+
+### API Routes
+
+Simply add an :extra-routes map to `make-untangled-server` with keys `:routes` and `:handlers`.
+`:routes` contains a [bidi](https://github.com/juxt/bidi) mapping from url route to a key in the `:handlers` map.
+`:handlers` is a mapping from handler key (from `:routes`) to a function (fn [env match] ... res)
+
+Eg:
+```
+(make-untangled-server
+  :extra-routes 
+  {:routes ["" {"/store-file" :store-file}]
+   :handlers {:store-file (fn [env match] (store-file (:db env) (get-in env [:request :body]))))})
+```
+
```

Comment:
How about a section on using ring middleware explicitly on a route instead of composing it into the stack? Or at least some cross-ref to something that talks more about this is in the ref guide or dev guide (tutorial)